src/browser/Decorations/OverviewRulerRenderer.ts

```diff
@@ -165,10 +186,11 @@ export class OverviewRulerRenderer extends Disposable {
 
   private _queueRefresh(updateCanvasDimensions?: boolean, updateAnchor?: boolean): void {
     if (this._animationFrame !== undefined) {
+      this._shouldUpdateDimensions = updateCanvasDimensions || this._shouldUpdateDimensions;
```

Comment:
This this needs to be done regardless of whether the animation frame exists yet or not:

```suggestion
  private _queueRefresh(updateCanvasDimensions?: boolean, updateAnchor?: boolean): void {
    this._shouldUpdateDimensions ||= updateCanvasDimensions;
    if (this._animationFrame !== undefined) {
```